src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/DebuggerDisplayInfo.cs

```diff
@@ -1,19 +1,80 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using Microsoft.VisualStudio.Debugger.Clr;
+using Microsoft.VisualStudio.Debugger.Evaluation;
 using Microsoft.VisualStudio.Debugger.Evaluation.ClrCompilation;
 
 namespace Microsoft.CodeAnalysis.ExpressionEvaluator
 {
-    internal struct DebuggerDisplayInfo
+    internal class DebuggerDisplayInfo
     {
+        private readonly DkmClrType m_targetType;
+        private DebuggerDisplayItemInfo m_value;
+        private DebuggerDisplayItemInfo m_simpleValue;
+
+        public DebuggerDisplayInfo(DkmClrType targetType)
+        {
+            m_targetType = targetType;
+        }
+
+        public bool HasValues { get { return (Name != null || TypeName != null || m_value != null); } }
+
+        public DebuggerDisplayItemInfo Name { get; private set; }
+
+        public DebuggerDisplayItemInfo TypeName { get; private set; }
+
+        public DebuggerDisplayItemInfo GetValue(DkmInspectionContext inspectionContext)
+        {
+            if (m_simpleValue != null && (inspectionContext.EvaluationFlags & DkmEvaluationFlags.UseSimpleDisplayString) == DkmEvaluationFlags.UseSimpleDisplayString)
+            {
+                return m_simpleValue;
+            }
+
+            return m_value;
+        }
+
+        internal void ApplyFavorities(DkmClrObjectFavoritesInfo favoritesInfo)
```

Comment:
>ApplyFavorities [](start = 22, length = 15)

We usually prefer immutable object. The pattern it WithFavoritesInfo(DkmClrObjectFavoritesInfo favoritesInfo). Please let me know if there is a reason to have it mutable.